src/provider/provider_os_memory.c

```diff
@@ -459,6 +529,18 @@ static umf_result_t os_alloc(void *provider, size_t size, size_t alignment,
         }
     }
 
+    if (os_provider->fd > 0) {
+        // store (fd_offset + 1) to be able to store fd_offset == 0
+        ret =
+            critnib_insert(os_provider->fd_offset_map, (uintptr_t)addr,
+                           (void *)(uintptr_t)(fd_offset + 1), 0 /* update */);
+        if (ret) {
+            LOG_ERR("os_alloc(): inserting a value to the IPC cache failed "
```

Comment:
soo.. this critnib is an IPC cache? perhaps the name of the variable is wrong...? or you could comment it somehow

Reply:
Replaced with "file descriptor offset map"

Reply:
thx!